Can edit puzzles

paymog opened the pull request:
Can edit everything property of a cryptogram except for its user or its
id

paymog committed:
Can edit puzzles

Can edit everything property of a cryptogram except for its user or its
id

paymog commented on src/tests/persistence/StubDB.java:
```diff
@@ -43,6 +43,83 @@ public boolean savePuzzle(Puzzle puzzle)
+		if (puz == null)
+		{
+			return false;
+		}
```
At this point you've probably noticed the same 4 lines being written over and over. Can you think of a better way to write this code to reduce the duplication?

rikimbo replied:
I can't think of anything that would easily reduce those lines, or factor them out.  I don't think something like this is that big of a deal, though.

willvoutier commented:
@paymog 

I would prefer a different (invisible) interface for updating. Since we're going to be re-using the creation screen, it'd be nice to not have to change it too much.

Do you think we could just do some `set[Attribute]()`'s on a puzzle that is already created, and then call `save()`?

The save would just need to figure out if the puzzle is new or updated (whether it has an `ID`).

Thoughts?

rikimbo commented:
I agree with what Will said.  That's how I'd think of doing it too, if I was implementing the edit feature.

paymog commented:
So what would the flow of logic be?

We select a puzzle
Press the edit button
Change stuff
Press save button -> this creates a new puzzle?
Save the newly created puzzle
At this point, the database (or PuzzleManager) should be able to detect a modified puzzle vs a new puzzle and act accordingly?

willvoutier commented:
Not quite. 

We select a puzzle
Press the edit button
Change stuff
Press save button -> the save will check to see if the puzzle has an ID or not (if it has one already, it'll update, otherwise it'll insert)

paymog committed:
Changed edit behaviour

The overall behaviour of save has changed.

When saving a puzzle with ID == DEFAULT_ID, the puzzle gets saved. If
the ID !=DEFAULT_ID then the PuzzleManager tries to find a puzzle with
the same ID and change its values.

So, to edit a puzzle, simply change it's values (except it's ID) and
then save it with the puzzlemanager.

paymog committed:
Deleted commented code

willvoutier commented:
Looks good.
